File: java/src/main/java/com/ccpayment/services/MerchantBatchWithdrawService.java
package com.ccpayment.services;

import com.ccpayment.APIError;
import com.ccpayment.Client;
import com.google.gson.JsonObject;

import java.io.IOException;

/**
 * Merchant batch withdraw service.
 * Request fields: taskName, orders[].orderId, orderIds, confirmExecution, stats.execOrderId.
 */
public class MerchantBatchWithdrawService {
    private final Client client;

    public MerchantBatchWithdrawService(Client client) {
        this.client = client;
    }

    public JsonObject checkWithdrawAddress(JsonObject data) throws APIError, IOException {
        return client.post("/checkWithdrawAddress", data, JsonObject.class);
    }

    public JsonObject applyBatchWithdraw(JsonObject data) throws APIError, IOException {
        return client.post("/applyBatchWithdraw", data, JsonObject.class);
    }

    public JsonObject appendBatchWithdraw(JsonObject data) throws APIError, IOException {
        return client.post("/appendBatchWithdraw", data, JsonObject.class);
    }

    public JsonObject confirmBatchWithdraw(JsonObject data) throws APIError, IOException {
        return client.post("/confirmBatchWithdraw", data, JsonObject.class);
    }

    public JsonObject abortBatchWithdraw(JsonObject data) throws APIError, IOException {
        return client.post("/abortBatchWithdraw", data, JsonObject.class);
    }

    public JsonObject getBatchWithdrawOrder(JsonObject data) throws APIError, IOException {
        return client.post("/getBatchWithdrawOrder", data, JsonObject.class);
    }

    public JsonObject getBatchWithdrawOrderRecordList(JsonObject data) throws APIError, IOException {
        return client.post("/getBatchWithdrawOrderRecordList", data, JsonObject.class);
    }
}

File: ruby/lib/ccpayment/services/merchant_batch_withdraw_service.rb
module CCPayment
  module Services
    class MerchantBatchWithdrawService
      def initialize(client)
        @client = client
      end

      def check_withdraw_address(data)
        @client.post('/checkWithdrawAddress', data)
      end

      def apply_batch_withdraw(data)
        @client.post('/applyBatchWithdraw', data)
      end

      def append_batch_withdraw(data)
        @client.post('/appendBatchWithdraw', data)
      end

      def confirm_batch_withdraw(data)
        @client.post('/confirmBatchWithdraw', data)
      end

      def abort_batch_withdraw(data)
        @client.post('/abortBatchWithdraw', data)
      end

      def get_batch_withdraw_order(data)
        @client.post('/getBatchWithdrawOrder', data)
      end

      def get_batch_withdraw_order_record_list(data)
        @client.post('/getBatchWithdrawOrderRecordList', data)
      end
    end
  end
end
